typed_sql/lib/src/adapter/sqlite_adapter.dart

```diff
@@ -417,6 +418,19 @@ final class _SqliteRowReader extends RowReader {
     return value;
   }
 
+  @override
+  JsonValue? readJsonValue() {
+    final value = jsonb.decode(_row.values[_i++] as Uint8List);
```

Comment:
Perhaps it's worthwhile catching the decoding exceptions here

Reply:
Adding addition type checks for `_row.values[_i++]`.

However, not sure about the `jsonb.decode` (or the later `.encode`) method, since it is an method inside the sqlite package, and it throws Errors on codec issues. What would expect here to do with them?

Reply:
> What would expect here to do with them?

Throw a database adapter specific exception, such that higher up we know where it comes from. We know that something weird is happening with the database adapter, or data in the database doesn't satisfy the schema. We know it's not something the user did wrong.